cpp/box/CMakeLists.txt

```
@@ -0,0 +1,22 @@
+# create the target and add the needed properties
+nanobind_add_module(_box SHARED module-box.cc export-box.h)
```

```suggestion
nanobind_add_module(_box NB_SHARED module-box.cc export-box.h)
```
I think we want `NB_SHARED` here per the discussion in https://nanobind.readthedocs.io/en/latest/api_cmake.html#high-level-interface . This avoids duplicating the nanobind compiled code in every freud module.